### ISSUE TITLE
System.ArgumentNullException: Value cannot be null in SyncService when no recent Workouts found

### ISSUE THREAD

**Issue by nbthor** [philosowaffle/peloton-to-garmin#552]
Consider checking out the [discussion forum](https://github.com/philosowaffle/peloton-to-garmin/discussions) to get help from the community.  Or fill out this template to get help here.

**Problem**
Cannot get the console to work properly. It stopped syncing about 1 week or so ago. I tried a couple of the interim fixes and finally 3.6.0. The last two iterations have thrown the attached error. I have tried both saving a configuration.local.json and creating a new one. I have double checked my login credentials. I set line 50 to "Debug", which shows a lot of information and then immediately closes the window. Playing around with the Configuration file, I am finding that if I set the line for the number of workouts to 0, this error occurs. If I put a number in that line, everything works like it should.

**Logs if applicable**
You can set your [config](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json#L50) to `"Debug"` to get more detailed logs.
```
2023-10-11 13:55:19.358 -04:00 [INF] *********************************************
2023-10-11 13:55:19.394 -04:00 [INF] P2G Version: p2g_console 3.6.0
2023-10-11 13:55:19.398 -04:00 [INF] Operating System: Win32NT
2023-10-11 13:55:19.400 -04:00 [INF] OS Version: Microsoft Windows NT 10.0.22635.0
2023-10-11 13:55:19.401 -04:00 [INF] DotNet Runtime: Microsoft Windows NT 10.0.22635.0
2023-10-11 13:55:19.401 -04:00 [INF] Docker Deployment: false
2023-10-11 13:55:19.403 -04:00 [INF] Config path: C:\Peloton to Garmin\configuration.local.json
2023-10-11 13:55:19.405 -04:00 [INF] *********************************************
2023-10-11 13:55:20.339 -04:00 [INF] *********************************************
2023-10-11 13:55:20.347 -04:00 [INF] *********************************************
2023-10-11 13:55:28.220 -04:00 [FTL] Uncaught Exception
System.ArgumentNullException: Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Where[TSource](IEnumerable`1 source, Func`2 predicate)
   at Sync.SyncService.FilterToCompletedWorkoutIds(ICollection`1 workouts) in D:\a\peloton-to-garmin\peloton-to-garmin\src\Sync\SyncService.cs:line 213
   at Sync.SyncService.SyncWithWorkoutLoaderAsync(Func`1 loader, ICollection`1 exclude) in D:\a\peloton-to-garmin\peloton-to-garmin\src\Sync\SyncService.cs:line 276
   at Sync.SyncService.SyncAsync(Int32 numWorkouts) in D:\a\peloton-to-garmin\peloton-to-garmin\src\Sync\SyncService.cs:line 56
   at PelotonToGarminConsole.Startup.RunAsync(CancellationToken cancelToken)
2023-10-11 14:08:11.097 -04:00 [INF] *********************************************
2023-10-11 14:08:11.146 -04:00 [INF] P2G Version: p2g_console 3.6.0
2023-10-11 14:08:11.155 -04:00 [INF] Operating System: Win32NT
2023-10-11 14:08:11.157 -04:00 [INF] OS Version: Microsoft Windows NT 10.0.22635.0
2023-10-11 14:08:11.158 -04:00 [INF] DotNet Runtime: Microsoft Windows NT 10.0.22635.0
2023-10-11 14:08:11.159 -04:00 [INF] Docker Deployment: false
2023-10-11 14:08:11.161 -04:00 [INF] Config path: C:\PelotontoGarmin\configuration.local.json
2023-10-11 14:08:11.161 -04:00 [INF] *********************************************
2023-10-11 14:08:12.333 -04:00 [INF] *********************************************
2023-10-11 14:08:12.386 -04:00 [INF] *********************************************
2023-10-11 14:08:22.002 -04:00 [FTL] Uncaught Exception
System.ArgumentNullException: Value cannot be null. (Parameter 'source')
   at System.Linq.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Linq.Enumerable.Where[TSource](IEnumerable`1 source, Func`2 predicate)
   at Sync.SyncService.FilterToCompletedWorkoutIds(ICollection`1 workouts) in D:\a\peloton-to-garmin\peloton-to-garmin\src\Sync\SyncService.cs:line 213
   at Sync.SyncService.SyncWithWorkoutLoaderAsync(Func`1 loader, ICollection`1 exclude) in D:\a\peloton-to-garmin\peloton-to-garmin\src\Sync\SyncService.cs:line 276
   at Sync.SyncService.SyncAsync(Int32 numWorkouts) in D:\a\peloton-to-garmin\peloton-to-garmin\src\Sync\SyncService.cs:line 56
   at PelotonToGarminConsole.Startup.RunAsync(CancellationToken cancelToken)

```

**Installation (please complete the following information):**
 - OS: Windows 11 Pro, 23H2 
 - Method:  exe
 - Version: 3.6.0

**Additional context**

**Comment by philosowaffle** [comment]
Could you share your config file? Remember to exclude any Garmin/Peloton credentials. 

> I set line 50 to "Debug", which shows a lot of information and then immediately closes the window.

When this happens the logs should (hopefully) also be written to `./output/logs.txt'

**Comment by nbthor** [comment]
Here's the configuration file. This one with "numWorkoutsToDownload" : 5, works fine. If I set it to 0, so that I can pick the number of workouts, it sends the error message above. No log file was generated by the "Debug" in line 50. Perhaps I did it wrong.

{
  "App": {
    "OutputDirectory": "./output",
    "EnablePolling": true,
    "PollingIntervalSeconds": 86400,
    "CloseWindowOnFinish": false,
    "CheckForUpdates": true
  },

  "Format": {
    "Fit": true,
    "Json": false,
    "Tcx": false,
    "SaveLocalCopy": true,
    "IncludeTimeInHRZones": false,
    "IncludeTimeInPowerZones": true,
    "DeviceInfoPath": "./deviceInfo.xml"
  },

  "Peloton": {
    "Email": "",
    "Password": "",
    "NumWorkoutsToDownload": 5,
    "ExcludeWorkoutTypes": [ "OutdoorWalking" ]
  },

  "Garmin": {
    "Email": "",
    "Password": "",
    "TwoStepVerificationEnabled": false,
    "Upload": true,
    "FormatToUpload": "fit",
    "UploadStrategy": 2
  },

  "Observability": {

    "Prometheus": {
      "Enabled": false,
      "Port": 4000
    },

    "Jaeger": {
      "Enabled": false,
      "AgentHost": "localhost",
      "AgentPort": 6831
    },

    "Serilog": {
      "Using": [ "Serilog.Sinks.Console", "Serilog.Sinks.File" ],
      "MinimumLevel": {
        "Default": "Information",
        "Override": {
          "Microsoft": "Error",
          "System": "Error"
        }
      },
      "WriteTo": [
        { "Name": "Console" },
        {
          "Name": "File",
          "Args": {
            "path": "./output/log.txt",
            "rollingInterval": "Day",
            "retainedFileCountLimit": 7
          }
        }
      ]
    }
  }
}

**Comment by LindsFisch** [comment]
I just downloaded the latest version and I'm having the same issue.  I'm happy to provide info if needed as well.

**Comment by ftischha** [comment]
I had a similar issue when I just tried to replace the files from 3.5 directly with the files from the 3.6 zip file.

It got solved when I deleted all the old files in the installation directory (except configuration.local.json) and then copied the new files from the 3.6 zip into it.

**Comment by philosowaffle** [comment]
Thanks for the additional details! This is on my radar, however I have family in town for the next week so it will be a bit before I can look into this.

**Comment by philosowaffle** [comment]
I believe I have this fixed now on the latest release [3.6.1](https://github.com/philosowaffle/peloton-to-garmin/releases/tag/v3.6.1). Sorry for the delay.
